### DISCUSSION TITLE
How to configure "java.test.config"?

### DISCUSSION THREAD

**Discussion by satoshiyamamoto** [mfussenegger/nvim-jdtls#505] [Q&A] [answered]
I would like to specify the following [JVM arguments](https://github.com/Microsoft/vscode-java-test/wiki/Run-with-Configuration) for a vscode-java-test installation.

```jsonc
"java.test.config": [
    {
        "name": "myConfiguration",
        "workingDirectory": "${workspaceFolder}",
        "vmargs": [ "-Xmx512M" ],
        "env": { "key": "value" },
    },
    {
        // Another configuration item...
    },
  ...
]
```

When adding configure in nvim-jdtls, is the following correct?
```lua
config.settings = {
   java = {
    ...
     test = {
       config = {
         {
           name = "testConfig",
           vmArgs = {
             "-Xmx512M",
             "--add-opens=java.base/java.lang=ALL-UNNAMED",
             "--add-opens=java.base/java.nio=ALL-UNNAMED",
           },
         },
       },
...

**Comment by gbroques** [comment]
I'm unable to configure `java.test.config` as well.

I tried the following in my Neovim configuration:
```lua
settings = {
  ['java.test.config'] = {
      vmArgs = {
        '-XX:+AllowRedefinitionToAddDeleteMethods'
      }
    }
},
```

The equivalent JSON configuration for VS Code is:
```json
"java.test.config": {
    "vmArgs": [
        "-XX:+AllowRedefinitionToAddDeleteMethods"
    ]
} 
```

Any thoughts @mfussenegger?

**Reply by mfussenegger** [reply, marked as answer]
There is no `java.test.config` setting for nvim-jdtls. `java.test.config` is a client setting specific to vscode-java.

The _debug adapter_ (java-debug) configuration option is just `vmArgs` (See `:help JdtDapConfig`)

You can either set that in your `dap.configurations.java` entries, or if you're using any of the jdtls debug functions you can set configuration overrides via `config_overrides` parameters.

See https://github.com/mfussenegger/nvim-jdtls/discussions/305#discussioncomment-3331169 for an example

**Reply by gbroques** [reply]
> ... if you're using any of the jdtls debug functions you can set configuration overrides via config_overrides parameters.

Thank you @mfussenegger!!

The following worked for me:

```lua
local java_test_config = { vmArgs = '-XX:+AllowRedefinitionToAddDeleteMethods' }
vim.keymap.set('n', '<leader>df', function() jdtls.test_class({ config_overrides = java_test_config }) end)
vim.keymap.set('n', '<leader>dn', function() jdtls.test_nearest_method({ config_overrides = java_test_config }) end)
```

**Reply by satoshiyamamoto** [reply]
Thank you all for your responses. It worked as expected in my environment!
